src/Component/Checker/InternalClock.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Jose\Component\Checker;
+
+use DateTimeImmutable;
+use Psr\Clock\ClockInterface;
+
+/**
+ * @internal
+ */
+final class InternalClock implements ClockInterface
+{
+    public function now(): DateTimeImmutable
+    {
+        return new DateTimeImmutable();
```

Comment:
I went with this instead of `DateTimeImmutable::createFromFormat(...)` because that could in theory return false (static analysis issue).
Could be handled differently of course, but this seems the most simple.

Reply:
Yes indeed, no need for microtime here.